### ISSUE TITLE
SFT OOM

### ISSUE THREAD

**Issue by 545487677** [OPPO-PersonalAI/Agent_Foundation_Models#5]
您好，非常感谢你们的工作，看了很有启发。
尝试在web_agent的数据集上进行sft，发现训练过程出现了OOM, 我的配置是单节点8卡A100 80G，然后BATCH_SIZES=(1)  
GRADIENT_ACCUMULATIONS=(16)， CUTOFF_LEN=32768，想问问需要怎么配置才能在8卡80G上面进行sft呢，谢谢。

**Comment by 545487677** [comment]
Update: 当我设置CUTOFF_LEN=16384， BATCH_SIZES=(1)  GRADIENT_ACCUMULATIONS=(4), 8卡A100 80G能够运行了，每张卡大概占50-60GB。  
0%|          | 0/684 [00:00<?, ?it/s]
  0%|          | 1/684 [00:36<6:57:41, 36.69s/it]
  0%|          | 2/684 [01:10<6:35:55, 34.83s/it]
  0%|          | 3/684 [01:43<6:29:15, 34.30s/it]
  1%|          | 4/684 [02:18<6:29:51, 34.40s/it]
  1%|          | 5/684 [02:55<6:40:08, 35.36s/it]
  1%|          | 6/684 [03:25<6:19:12, 33.56s/it]
  1%|          | 7/684 [03:56<6:07:35, 32.58s/it]
  1%|          | 8/684 [04:37<6:38:49, 35.40s/it]
  1%|▏         | 9/684 [05:07<6:17:21, 33.54s/it]
  1%|▏         | 10/684 [05:40<6:17:36, 33.61s/it]
                                                  
{'loss': 1.3586, 'grad_norm': 5.288890145459093, 'learning_rate': 1.8260869565217391e-06, 'epoch': 0.04}，
就是不知道这样缩减了长度，会不会对效果造成影响.....

**Comment by chenqianben** [comment]
你好，请问你用的是多大尺寸的模型呢？一般我们SFT训练的时候，cutoff_len=32768是可以双机八卡跑起来的；cutoff_len=16384是会影响效果的

**Comment by 545487677** [comment]
我是qwen2.5-7b-instruct来跑的，只有单机8卡...

**Comment by Tianyue818** [comment]
可以参考一下这个项目：https://github.com/Qihoo360/360-LLaMA-Factory。
对OOM问题可能有一些帮助。🙂
